src/transformStateWithClones.js

```diff
@@ -8,6 +8,27 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const newObj = Object.assign({}, state);
+  const result = [];
+
+  for (let prop = 0; prop < actions.length; prop++) {
+    if (actions[prop].type === 'addProperties') {
+      Object.assign(newObj, actions[prop].extraData);
+      result.push({ ...newObj });
+    } else if (actions[prop].type === 'removeProperties') {
+      for (const key of actions[prop].keysToRemove) {
+        delete newObj[key];
+      }
+      result.push({ ...newObj });
+    } else if ((actions[prop].type === 'clear')) {
+      for (const key in newObj) {
+        delete newObj[key];
+      }
+      result.push({});
+    }
```

Comment:
avoid such nesting `else if` conditions
This code is hard to read and maintain
use `switch` instead

Reply:
Fixed it, thanks for the advice!